File: 1049.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
var ossos = lines.shift();
var animal = lines.shift();
var alimento = lines.shift();
if (ossos === 'vertebrado'){
    if (animal === 'ave'){
        if (alimento === 'carnivoro'){
            console.log('aguia');}
        else if (alimento === 'onivoro'){
            console.log('pomba')}}
    else if (animal === 'mamifero'){
        if (alimento === 'onivoro'){
            console.log('homem')}
        else if (alimento === 'herbivoro'){
            console.log('vaca')}}}

else if (ossos === 'invertebrado'){
    if (animal === 'inseto'){
        if (alimento === 'hematofago'){
            console.log('pulga')}
        else if (alimento === 'herbivoro'){
            console.log('lagarta')}}
    else if (animal === 'anelideo'){
        if (alimento === 'hematofago'){
            console.log('sanguessuga')}
        else if (alimento === 'onivoro'){
            console.log('minhoca')}}}

File: 1049.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main(){
    string ossos, animal, alimento;
    cin >> ossos;
    cin >> animal;
    cin >> alimento;
    if(ossos == "vertebrado"){
        if(animal == "ave"){
            if(alimento == "carnivoro"){
                cout << "aguia" << "\n";
            }
            else if(alimento == "onivoro"){
                cout << "pomba" << "\n";
            }
        }
        else if(animal == "mamifero"){
            if(alimento == "onivoro"){
                cout << "homem" << "\n";
            }
            else if(alimento == "herbivoro"){
                cout << "vaca" << "\n";
            }
        }
    }
    else if(ossos == "invertebrado"){
        if(animal == "inseto"){
            if(alimento == "hematofago"){
                cout << "pulga" << "\n";
            }
            else if(alimento == "herbivoro"){
                cout << "lagarta" << "\n";
            }
        }
        else if(animal == "anelideo"){
            if(alimento == "hematofago"){
                cout << "sanguessuga" << "\n";
            }
            else if(alimento == "onivoro"){
                cout << "minhoca" << "\n";
            }
        }
    }
    return 0;
}
